openapi-core/src/main/java/io/ballerina/openapi/core/generators/service/ReturnTypeGenerator.java

```diff
@@ -133,11 +133,14 @@ public ReturnTypeDescriptorNode getReturnTypeDescriptorNode(Map.Entry<PathItem.H
                 //handle multiple response scenarios ex: status code 200, 400, 500
                 TypeDescriptorNode type = handleMultipleResponse(responses);
                 returnNode = createReturnTypeDescriptorNode(createToken(RETURNS_KEYWORD), annotations, type);
-            } else {
+            } else if (responses.size() == 1) {
```

Comment:
Can a response have `null` as a value?

Reply:
An endpoint in a OpenAPI contract needs to have the `responses:` sections. Otherwise it gives below error from the OpenAPI parser 

```
io.ballerina.openapi.core.exception.BallerinaOpenApiException: OpenAPI definition has errors: 
attribute paths.'/pets'(get).responses is missing
```

But as given in this issue there can be occasions where the `responses:` section has defined but its empty. (`responses : {}`). In such cases in the we get an empty map for responses.

Ex: `ApiResponses responses = operation.getValue().getResponses();`

here response is a `LinkedHashMap` with size zero. 

Could not find a occasion where we get `responses == null`